index.js

```diff
@@ -926,6 +927,109 @@ PersonaClient.prototype.getProfileByGuid = function(opts, callback){
     });
     req.end();
 };
+
+/**
+* Get all profiles for an array of guids.
+ *
+ * @param  {object}   opts     [description]
+ * @param  {array}    opts.guids  Array of GUIDs to fetch profiles for
+ * @param  {string}   opts.token  Auth token
+ * @param  {string}   opts.xRequestId Optional request ID to pass through in logging.
+ * @param  {Function} callback
+ */
+PersonaClient.prototype.getProfilesForGuids = function(opts, callback) {
+    validateOpts(opts,{guids: _.isArray,token: _.isString});
+    var guids = opts.guids;
+    var token = opts.token;
+    var xRequestId = opts.xRequestId || uuid.v4();
+    var personaQueryLimit = 25;
```

Comment:
If a limit does need applying I'd rather this was controlled via the usual limit and offset params at the API level, and the client respect that (as in, the caller is aware that they need to call this X number of times).